src/sentry/integrations/slack/utils.py

```diff
@@ -424,16 +426,31 @@ def get_channel_id_with_timeout(integration, name, timeout):
                 )
                 return (prefix, None, False)
 
-            item_id = {c["name"]: c["id"] for c in items[result_name]}.get(name)
-            if item_id:
-                return (prefix, item_id, False)
+            for c in items[result_name]:
+                # The "name" field is unique (this is the username for users)
+                # so we return immediately if we find a match.
+                if c["name"] == name:
+                    return (prefix, c["id"], False)
+                # If we don't get a match on a unique identifier, we look through
+                # the users' display names, and error if there is a repeat.
+                if list_type == "users":
+                    profile = c.get("profile")
+                    if profile and profile.get("display_name") == name:
+                        if id_data:
+                            found_duplicate = True
+                        else:
+                            id_data = (prefix, c["id"], False)
 
             cursor = items.get("response_metadata", {}).get("next_cursor", None)
             if time.time() > time_to_quit:
                 return (prefix, None, True)
 
             if not cursor:
                 break
+        if found_duplicate:
+            raise IntegrationError("Duplicate display name.")
```

Comment:
Does it make more sense to raise as soon as we find a duplicate, rather than here? Then we can skip having `found_duplicate` entirely.

Reply:
That was what I had originally, but I think we still want to return a success if we match on a unique identifier (i.e. username), which could theoretically happen after a duplicate match on display name. Certainly not likely, but still possible.